### PR TITLE
fix:修复输入框被键盘挡住的问题

### LINKED ISSUE #2737: InputItem的type为money时，如果InputItem在最底部，将会被完全遮挡，有何解决办法？
- [x] I have searched the [issues](https://github.com/ant-design/ant-design-mobile/issues) of this repository and believe that this is not a duplicate.

### Version
undefined

### Environment
IOS11.4.1，chrome最新版

### Reproduction link
[http://antd-mobile.gitee.io/components/input-item-cn/](http://antd-mobile.gitee.io/components/input-item-cn/)

### Steps to reproduce
把InputItem放到界面最底部，设置它的type为money时，InputItem将会被money键盘完全遮挡

### What is expected?
像普通键盘那样输入框不被遮挡

### What is actually happening?
money键盘遮挡了输入框

### PR DESCRIPTION
**Pull request by likezero**
add a new option autoAdjustHeight  `for type=money`

* [ x ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ x ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ x ] Rebase before creating a PR to keep commit history clear.
* [ x ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ x ] Update API docs for the component.
  * [ x ] Update/Add demo to demonstrate new feature.
  * [ x ] Update TypeScript definition for the component.
  * [ x ] Add unit tests for the feature.

close #2737

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3243)

### CODE REVIEW

File: components/input-item/CustomInput.tsx [hunk 1]
```diff
@@ -241,11 +244,71 @@ class NumberInput extends React.Component<NumberInputProps, any> {
...
   }
+  fixedBody = () => {
+    // 保险起见，让用户主动开启。键盘的使用场景比较复杂。
+    if (!this.props.autoAdjustHeight) {
+      return;
+    }
+    const keyBoardWrappers = document.querySelectorAll(`.${this.props.keyboardPrefixCls}-wrapper`);
+    const { focus } = this.state;
+    // 取最后一个
+    const keyBoardWrapper = keyBoardWrappers[keyBoardWrappers.length - 1];
```
Comment (ziluo) [inline components/input-item/CustomInput.tsx]: 这里为啥不用Ref来获取呢，有啥特殊逻辑么

File: components/input-item/CustomInput.tsx [hunk 2]
```diff
@@ -241,11 +244,71 @@ class NumberInput extends React.Component<NumberInputProps, any> {
...
+              this.keyboardHeight = parseInt(keyboardHeight, 10) || 0;
+              // // 重设paddingBottom让页面整体向上抬
+              document.body.style.paddingBottom = (parseInt(paddingBottom, 10) + this.keyboardHeight) + 'px';
+              // // 页面滚动
+              document.body.scrollTop = document.body.scrollTop + this.keyboardHeight;
+            }
+          }
+        }, 300);
+      } else {
+        // 键盘已经被隐藏，恢复
```
Comment (ziluo) [inline components/input-item/CustomInput.tsx]: 恢复padding的逻辑抽取一下吧，两个地方同样的代码

File: components/input-item/CustomInput.tsx [hunk 3]
```diff
@@ -241,11 +244,71 @@ class NumberInput extends React.Component<NumberInputProps, any> {
...
+      if (focus) {
+        // 当重复操作当前输入框时，键盘状态循环变化：show-hide-show-hide...
+        // 当键盘处于hide态时，样式设置了transition-duration=0.2s。会导致样式获取不到最新的。因此延时300ms获取正常的值。
+        setTimeout(() => {
+          const wrapperOffset = keyBoardWrapper.getBoundingClientRect();
+          if (this.inputRef) {
+            const inputOffset = this.inputRef.getBoundingClientRect();
+            // 输入框被键盘挡着
+            if (inputOffset.top > wrapperOffset.top) {
+              const bodyStyles = window.getComputedStyle(document.body);
```
Comment (ziluo) [inline components/input-item/CustomInput.tsx]: getBoundingClientRect 和 getComputedStyle 要不要考虑下兼容性问题？

**Comment by zbth5** [comment]
在ios上autoAdjustHeight这个属性没有用？需要怎么解决呢？

**Pull request closed without merging**